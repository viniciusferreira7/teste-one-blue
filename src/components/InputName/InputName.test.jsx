import { render, screen } from '@testing-library/react';
import { InputName } from '.';

describe('<InputName/>', () => {
  it('should render text in the placeholder', () => {
    render(<InputName text="Usuário" />);
    expect(screen.getByPlaceholderText('Usuário')).toBeInTheDocument();
  });

  it('should render text in the placeholder', () => {
    render(<InputName name="name" text="Usuário" />);
    expect(screen.getByPlaceholderText('Usuário')).toHaveAttribute(
      'name',
      'name',
    );
  });

  it('should have attribute required', () => {
    render(<InputName text="Usuário" />);
    expect(screen.getByPlaceholderText('Usuário')).toHaveAttribute('required');
  });

  it('should correct style', () => {
    render(<InputName name="name" text="Usuário" />);
    expect(screen.getByPlaceholderText('Usuário')).toHaveStyle(
      'font-size: 1.6rem',
    );
  });
});
