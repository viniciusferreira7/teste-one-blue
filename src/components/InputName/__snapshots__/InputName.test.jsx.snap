// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputName/> should match snapshot 1`] = `
<input
  class="sc-bczRLJ drTqQa"
  name="name"
  placeholder="UsuÃ¡rio"
  required=""
  type="text"
/>
`;
